packages/blade/src/components/InputGroup/InputRow.web.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import styled from 'styled-components';
+import type { InputRowProps } from './types';
+import { assignWithoutSideEffects } from '~utils/assignWithoutSideEffects';
+
+const StyledInputRow = styled.div<{ $gridTemplateColumns: string }>`
+  display: grid;
+  grid-template-columns: ${(props) => props.$gridTemplateColumns};
+`;
+
+// Helper function to recursively add _inputPosition to all children
+const addInputPositionToChildren = (
```

Comment:
Why use all the complicated React children hacks instead of simple CSS? 

https://codesandbox.io/p/sandbox/2qkhdw

Reply:
We do similar stuff even on [ButtonGroup](https://github.com/razorpay/blade/blob/a72944305f0f1374b50a25e8ef7abb52b81fe792/packages/blade/src/components/ButtonGroup/StyledButtonGroup.tsx#L22-L34)